### PR TITLE
Detect embedded and skip addons

### PR DESCRIPTION
**Pull request by humitos**
Check if addons library is being loaded inside an iframe, and skip loading it.

Closes https://github.com/readthedocs/addons/issues/412

### CODE REVIEW

**Comment by humitos** [comment]
> I think this should be an option.

Hrm, good point. I don't think we should expose this to users, since it seems to be pretty complex/advanced/confusing and probably not useful for 99% of our users. However, I think we should probably have an option internally for this and enable/disable it via support request.

> We have at least one customer on .com that makes use of iframes, where it does need to show the flyout.

Do they have public docs where I can see how they are using iframes?

**Comment by stsewd** [comment]
> Do they have public docs where I can see how they are using iframes?

If I remember correctly, their use case was to load their private documentation using a sharing token from an iframe, probably from inside their application.

> However, I think we should probably have an option internally for this and enable/disable it via support request.

We can also expose an option as a query parameter, since the parent page should be the one that decides if the flyout should be shown. This is since users can embed docs they don't have control over. I imagine something like `?rtd-disable-flyout`, users embedding docs and that don't want to show the flyout there can include the page with that query.

**Comment by humitos** [comment]
> This is since users can embed docs they don't have control over.

I'm not too worried about this case.

> I imagine something like `?rtd-disable-flyout`, users embedding docs and that don't want to show the flyout there can include the page with that query.

I think we want to depend on the user for this. How this approach will solve the current problem we have? The user will need to rebuild all their documentation and probably modify the Sphinx extension's code to add that query attribute --which seems pretty complex for regular users.

I think the general rule we were using makes sense, since it follows what we used to have:

- if it's embed, don't enable addons
- if it's embed and there is a config that force addons, enable adons
- if it's not embed, always enable addons

That "config that forces addons" will be a META tag as we are doing with other things we want to communicate from the backend, and it will be a field in `AddonsConfig` so it can be manged per project. It won't be exposed to users and managed via support request.

**Comment by stsewd** [comment]
> I think we want to depend on the user for this. How this approach will solve the current problem we have? The user will need to rebuild all their documentation and probably modify the Sphinx extension's code to add that query attribute --which seems pretty complex for regular users.

They just need to update the target URL, that's not complex, they are already providing a URL. And that doesn't require users to ask us to disable the option on docs they don't own.

**Comment by humitos** [comment]
> I think we want to depend on the user for this.

I missed typed this. I meant here _I think we **don't** want to depend on the user for this_.

**Comment by humitos** [comment]
I will add an option in the server that by default skips loading the addons in iframes, but that we can manually force injecting addons for those users requiring this feature.

**Review by agjohnson — APPROVED** [review]

**Review by agjohnson — APPROVED** [review]

File: src/index.js [hunk 1]
```diff
@@ -44,6 +46,15 @@ export function setup() {
         return getReadTheDocsConfig(sendUrlParam);
       })
       .then((config) => {
+        const loadWhenEmbedded = objectPath.get(
+          config,
+          "addons.configs.load_when_embedded",
```
[the remark this answers is not in the document]
Comment (humitos) [reply src/index.js:51]: I changed it by `options`.

**Pull request merged**